morebits.js

```diff
@@ -4566,215 +4577,1725 @@ Morebits.wiki.page = function(pageName, status) {
+				// Likewise, save the widest rangeblock.
+				// Could also pre-sort this by expiry, as that may be more useful.
+				var block, subnet = 0;
+				response.blocks.reverse().forEach(function(bl, idx) {
+					if (bl.user === ctx.userName || (idx === response.blocks.length - 1 && !block)) {
+						block = bl;
+					}
+					// Always false (0.0.0.0 === 0.0.0.0) for users
+					// Ensure we get the largest range
+					if (bl.rangestart !== bl.rangeend && (!subnet || bl.user.split('/')[1] < subnet)) {
+						subnet = bl.user.split('/')[1];
+						ctx.isRangeBlocked = true;
+						ctx.blockedRange = bl.user;
+					}
+				});
+				// blockInfo object used by getters
+				ctx.blockInfo = $.extend({}, block);
+
+				// If this is the actual user in question, override the default
+				// context values in order to default a reblock to the existing parameters.
+				if (ctx.blockInfo.user === ctx.userName && ctx.useOriginalBlockParams) {
+					// Note that expiry and reason aren't here,
+					// as they can apply to non-block methods;
+					// they are handled in fnProcessBlock.
+					['allowusertalk', 'anononly', 'autoblock', 'nocreate', 'noemail', 'partial'].forEach(function(param) {
+						ctx[param] = !!block[param];
+					});
+
+					// hidename, not hidden, since when applying a block, it's hidename.
+					// See also user.hidden aka ctx.hidden.
+					ctx.hidename = !!block.hidden;
+
+					if (ctx.partial) {
+						if (block.restrictions.namespaces) {
+							ctx.namespacerestrictions = block.restrictions.namespaces;
+						}
+						// Force into array of titles, ditch ns (included in title) and page ID
+						if (block.restrictions.pages) {
+							ctx.pagerestrictions = block.restrictions.pages.map(function(rp) {
+								return rp.title;
+							});
+						}
+					}
+				}
+			}
+		} else {
+			// User doesn't exist locally
+			// Suppressed (and gsuppressed) names show
+			// up here as well to those without the permission.
+			// In the future, could consider adding cancreate to
+			// usprop if we wanted to allow for account creation
+			// Which apparently is fucking hard https://www.mediawiki.org/wiki/API:Account_creation#Creating_an_account
+			ctx.userName = '';
+		}
 
-	var level = []; // Track of how deep we are ({{, {{{, or [[)
-	var count = -1;  // Number of parameters found
-	var unnamed = 0; // Keep track of what number an unnamed parameter should receive
-	var equals = -1; // After finding "=" before a parameter, the index; otherwise, -1
-	var current = '';
-	var result = {
-		name: '',
-		parameters: {}
+		// Talk page stuff
+		// Ignore unresolved, invalid page titles (e.g. circular redirects)
+		var page = response.pages && response.pages[0];
+		if (page && !page.invalid) {
+			ctx.talkExists = !page.missing;
+			if (ctx.talkExists) {
+				// Update to redirect target or normalized name;
+				// no status message so as to avoid duplication when notifying
+				ctx.talkTitle = page.title;
```

Comment:
Originally attempted to notify on redirection here, like in mb.w.page, but would mean duplication of that message when notifying.  Maybe not a big deal to leave off here?